### DISCUSSION TITLE
Can't apply default values to protected properties

### DISCUSSION THREAD

**Discussion by fabricioffa** [livewire/livewire#8918] [1. Help]
I'm creating a base component for lazy ones to extend. As I didn't want some properties to appear on the front or change, I've set them as protected. But when I extend the component and don't pass values for these protected properties, i get an error despite them having a default value.

The base component: 

```
abstract class BaseLazyComponent extends Component
{
    use ApiRequest;
    public bool $isPlaceholder = false;
    public string $rootClasses;
    public string | array $view;
    protected array $query = [];
    protected int $timeout = 2;
    protected array $endpoints;
    protected array | string $events = [];

    public function mount($query, $timeout, $endpoints, $events)
    {
        $this->query = $query ?? [];
        $this->timeout = $timeout ?? 2;
        $this->endpoints = $endpoints;
        $this->events = $events ?? [];
    }
...
```

```
The extending component: 

class AvaiableBrands extends BaseLazyComponent
{
    public array $brands;

    public function mount($query, $timeout, $endpoints, $events)
    {
        parent::mount($query, $timeout, $endpoints, $events);
        $this->brands = $this->fetchData()?->data ?? [];
    }
}
```
The component call: 

```
<livewire:avaiable-brands
            rootClasses="bg-gray-100 dark:bg-gradient-to-b dark:from-dark dark:via-dark-300 dark:to-dark py-10"  :
            :endpoints="['brands' => 'grouped/brands']"
            view="livewire.avaiable-brands"
            events="init-avaiable-brands-swiper"
            lazy
     />
```

With this call, i get the following error: 

```
Illuminate\Contracts\Container\BindingResolutionException

Unable to resolve dependency [Parameter #0 [ <required> $query ]] in class App\Livewire\AvaiableBrands
```

It fails on: 

`vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php : 188`

Could someone tell me what am I missing here?

**Comment by EdgarsJoja** [comment]
`mount` method in Livewire acts sort of as constructor. So you can pass (inject) dependencies there. You have specified that you will inject dependencies `$query`, `$timeout`, `$endpoints` and `$events`. You pass `$endpoints` and `$events`:
```blade
            :endpoints="['brands' => 'grouped/brands']"
            view="livewire.avaiable-brands"
            events="init-avaiable-brands-swiper"
```

but not `$query` and `$timeout`. And this is what the error is about - the code cannot resolve those dependencies, because nothing is passed.

Either pass query and timeout similarly as other, or assign default values in mount, or remove them from mount.
